starknet_py/tests/e2e/deploy/deploy_test.py

```diff
@@ -106,3 +108,126 @@ async def test_constructor_without_arguments(gateway_account_client):
     contract = result.deployed_contract
 
     assert contract.address is not None
+
+
+@pytest.mark.asyncio
+async def test_default_deploy_with_class_hash(
+    deployer_address, account_client, map_class_hash
+):
+    res = await account_client.deploy_contract(
+        class_hash=map_class_hash, deployer_address=deployer_address, max_fee=MAX_FEE
+    )
+
+    assert isinstance(res, int)
+
+
+@pytest.mark.asyncio
+async def test_throws_when_deployer_address_not_specified_on_custom_network(
+    account_client, map_class_hash
+):
+    with pytest.raises(ValueError) as err:
+        await account_client.deploy_contract(class_hash=map_class_hash, max_fee=MAX_FEE)
+
+    assert "deployer_address is required when not using predefined networks." in str(
+        err.value
+    )
+
+
+@pytest.mark.asyncio
+async def test_throws_when_constructor_calldata_without_abi(
+    account_client, map_class_hash, deployer_address
+):
+    with pytest.raises(ValueError) as err:
+        await account_client.deploy_contract(
+            class_hash=map_class_hash,
+            deployer_address=deployer_address,
+            constructor_calldata=[12, 34],
+            max_fee=MAX_FEE,
+        )
+
+    assert "constructor_calldata was provided without an abi" in str(err.value)
+
+
+@pytest.mark.asyncio
+async def test_throws_when_constructor_calldata_not_provided(
+    account_client, deployer_address
+):
+    compiled_contract = create_compiled_contract(
+        compilation_source=constructor_with_arguments_source
+    )
+    abi = compiled_contract.abi
+
+    with pytest.raises(ValueError) as err:
+        await account_client.deploy_contract(
+            class_hash=1234, abi=abi, deployer_address=deployer_address, max_fee=MAX_FEE
+        )
+
+    assert "Provided contract has a constructor and no args were provided." in str(
+        err.value
+    )
+
+
+@pytest.mark.asyncio
+async def test_constructor_arguments_deploy_contract(
+    account_client, new_gateway_account_client, deployer_address
+):
+    # pylint: disable=too-many-locals
+    abi = (
+        create_compiled_contract(compilation_source=constructor_with_arguments_source)
+    ).abi
+    class_hash = (
+        await new_gateway_account_client.declare(
+            await new_gateway_account_client.sign_declare_transaction(
+                compilation_source=constructor_with_arguments_source, max_fee=MAX_FEE
+            )
+        )
+    ).class_hash
+
+    value = 10
+    tuple_value = (1, (2, 3))
+    arr = [1, 2, 3]
+    struct = {"value": 12, "nested_struct": {"value": 99}}
```

Comment:
Can't we parametrize the test with these?

Reply:
There will be too many arguments in the test function